shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -4219,42 +4218,98 @@ module MakeImpl<InputSig Lang> {
         )
       }
 
-      private PathNodeImpl localStepToHidden(PathNodeImpl n) {
+      private PathNodeImpl localStep(PathNodeImpl n) {
         n.getASuccessorImpl() = result and
-        result.isHidden() and
         exists(NodeEx n1, NodeEx n2 | n1 = n.getNodeEx() and n2 = result.getNodeEx() |
           localFlowBigStep(n1, _, n2, _, _, _, _) or
           storeEx(n1, _, n2, _, _) or
           readSetEx(n1, _, n2)
         )
       }
 
+      private PathNodeImpl summaryCtxStep(PathNodeImpl n) {
+        n.getASuccessorImpl() = result and
+        exists(SummaryCtxSome sc |
+          pathNode(n, _, _, _, pragma[only_bind_into](sc), _, _, _) and
+          pathNode(result, _, _, _, pragma[only_bind_into](sc), _, _, _)
+        )
+      }
+
+      private PathNodeImpl localStepToHidden(PathNodeImpl n) {
+        result = localStep(n) and
+        result.isHidden()
+      }
+
+      private PathNodeImpl localStepFromHidden(PathNodeImpl n) {
+        n = localStep(result) and
+        result.isHidden()
```

Comment:
From the name I'd expect that `n` was hidden, not that the order was swapped.
```suggestion
      private PathNodeImpl localStepFromHidden(PathNodeImpl n) {
        result = localStep(n) and
        n.isHidden()
```

Reply:
Or maybe just change the predicate name:
```suggestion
      private PathNodeImpl revLocalStepToHidden(PathNodeImpl n) {
        n = localStep(result) and
        result.isHidden()
```

Reply:
Or maybe that should be `revLocalStepFromHidden`. I guess that depends on whether you read it as `rev` applying to all of the rest or just to `localStep`.

Reply:
I think the code would be clearer if both of these relations were using named columns instead of the `result` column and were both in the forward direction, i.e. `localStepFromHidden(n1, n2) and `localStepToHidden(n1, n2)`. This would add a few lines and name a few more intermediate nodes in the usages below, but I think that's a readability win.